edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs

```diff
@@ -265,6 +258,12 @@ public static async Task<int> MainAsync(IConfiguration configuration)
                         throw new InvalidOperationException($"ConfigSource '{configSourceConfig}' not supported.");
                 }
 
+                metricsConfig = new MetricsConfig(experimentalFeatures.EnableMetrics, MetricsListenerConfig.Create(configuration));
```

Comment:
Previously NullMetricProvider is registered if metric is not enabled. Why always register metric module?

Reply:
Previously I was using varun's method of storing the metrics provider in a global static class. Now the provider uses dependency injection into classes. When I moved the construction into module I thought all metrics disabled themselves, which made the null metrics provider unnecessary. However, I just realized the metrics command factory doesn't , so I'll add the null provider back in.